core/src/main/scala/org/apache/spark/deploy/DependencyUtils.scala

```diff
@@ -137,16 +137,29 @@ private[deploy] object DependencyUtils {
   def resolveGlobPaths(paths: String, hadoopConf: Configuration): String = {
     require(paths != null, "paths cannot be null.")
     Utils.stringToSeq(paths).flatMap { path =>
-      val uri = Utils.resolveURI(path)
-      uri.getScheme match {
-        case "local" | "http" | "https" | "ftp" => Array(path)
-        case _ =>
-          val fs = FileSystem.get(uri, hadoopConf)
-          Option(fs.globStatus(new Path(uri))).map { status =>
-            status.filter(_.isFile).map(_.getPath.toUri.toString)
-          }.getOrElse(Array(path))
+      val spath = path.split('#')
```

Comment:
Why not use `Utils.resolveURI` as before? Parsing URIs by hand is very sketchy.

Reply:
You are right. It took some time to clone a URI without the fragment part though but next version will include that.